### PR TITLE
use compare-and-swap instead of channel in exclusiveLock

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: libkbfs/leveled_mutex.go [hunk 1]
```diff
@@ -24,27 +25,23 @@ import (
...
-	case l.ch <- struct{}{}:
-	default:
+	if !atomic.CompareAndSwapInt32(l.v, 0, 1) {
 		panic("unexpected concurrent access")
 	}
 }
 
 func (l exclusiveLock) unlock() {
-	<-l.ch
+	atomic.StoreInt32(l.v, 0)
```
Comment (akalin-keybase) [inline libkbfs/leveled_mutex.go]: can you make this a CAS operation too? (from 1 to 0) That way we'd catch double-unlocks. (The old version should have done something similar, too.)

**Comment by akalin-keybase** [comment]
just one comment

**Comment by songgao** [comment]
Thanks @akalin-keybase! I'll hit merge after tests pass?

**Comment by akalin-keybase** [comment]
yeap, lgtm!

**Pull request merged**